src/plugincontainer.cpp

```diff
@@ -769,6 +781,193 @@ const PreviewGenerator &PluginContainer::previewGenerator() const
   return m_PreviewGenerator;
 }
 
+void PluginContainer::simulateStartup(const std::vector<IPlugin*>& plugins) const
+{
+  // setUserInterface()
+  if (m_UserInterface) {
+    for (auto* plugin : plugins) {
+      if (auto* proxy = dynamic_cast<IPluginProxy*>(plugin)) {
+        proxy->setParentWidget(m_UserInterface->mainWindow());
+      }
+      if (auto* modPage = dynamic_cast<IPluginModPage*>(plugin)) {
+        modPage->setParentWidget(m_UserInterface->mainWindow());
+      }
+      if (auto* tool = dynamic_cast<IPluginTool*>(plugin)) {
+        tool->setParentWidget(m_UserInterface->mainWindow());
+      }
+      if (auto* installer = dynamic_cast<IPluginInstaller*>(plugin)) {
+        installer->setParentWidget(m_UserInterface->mainWindow());
+      }
+    }
+  }
+
+  // Simulate callbacks, e.g. onUserInterfaceInitialized and onProfileChanged.
+  for (auto* plugin : plugins) {
+    auto* oproxy = organizerProxy(plugin);
+    oproxy->m_UserInterfaceInitialized(m_UserInterface->mainWindow());
+    oproxy->m_ProfileChanged(nullptr, m_Organizer->currentProfile());
+  }
+}
+
+void PluginContainer::loadPlugin(QString const& filepath)
+{
+  std::vector<IPlugin*> plugins;
+  if (QFileInfo(filepath).isFile() && QLibrary::isLibrary(filepath)) {
+    std::unique_ptr<QPluginLoader> pluginLoader(new QPluginLoader(filepath, this));
+    if (pluginLoader->instance() == nullptr) {
+      log::error("failed to load plugin {}: {}", filepath, pluginLoader->errorString());
+    }
+    else {
+      if (IPlugin* plugin = registerPlugin(pluginLoader->instance(), filepath, nullptr)) {
+        log::debug("loaded plugin '{}' from '{}' - [{}]",
+          plugin->name(), QFileInfo(filepath).fileName(), implementedInterfaces(plugin).join(", "));
+        m_PluginLoaders.push_back(pluginLoader.release());
+        plugins.push_back(plugin);
+      }
+      else {
+        m_FailedPlugins.push_back(filepath);
+        log::warn("plugin \"{}\" failed to load (may be outdated)", filepath);
+      }
+    }
+  }
+  else {
+    // We need to check if this can be handled by a proxy.
+    for (auto* proxy : this->plugins<IPluginProxy>()) {
+      auto filepaths = proxy->pluginList(
+        QCoreApplication::applicationDirPath() + "/" + ToQString(AppConfig::pluginPath()));
+      if (filepaths.contains(filepath)) {
+        plugins = loadProxied(filepath, proxy);
+        break;
+      }
+    }
+  }
+
+  for (auto* plugin : plugins) {
+    emit pluginRegistered(plugin);
+  }
+
+  simulateStartup(plugins);
+}
+
+void PluginContainer::unloadPlugin(MOBase::IPlugin* plugin, QObject* object)
+{
+  if (auto* game = qobject_cast<IPluginGame*>(object)) {
+    unregisterGame(game);
+  }
+
+  // We need to do this BEFORE unloading from the proxy otherwise the

```

Comment:
Can you clarify what "this" is?

Reply:
It's removing the plugins from the `m_Plugins` member (except the `QObject` since we are iterating over-it). I've clarified the comment.